### PR TITLE
[explore] improve metric(s) and groupby(s) controls

### PR DESCRIPTION
**Pull request by mistercrunch**
- surface verbose_name, description & expression in controls
- [table viz] surface verbose name in table header

screen shot 2017-06-06 at 11 33 15 pm

### CODE REVIEW

File: superset/assets/javascripts/explore/components/ChartContainer.jsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ class ChartContainer extends React.PureComponent {
     return {
       viewSqlQuery: this.props.queryResponse.query,
       containerId: props.containerId,
+      datasource: props.datasource,
```
Comment (ascott) [inline superset/assets/javascripts/explore/components/ChartContainer.jsx]: curious why we are using `this.props` and `props` in this method. would be nice to clean this up and use either `props` or `this.props` for consistency.

File: superset/assets/javascripts/explore/components/controls/SelectControl.jsx [hunk 2]
```diff
@@ -27,6 +31,9 @@ const defaultProps = {
   multi: false,
   onChange: () => {},
   showHeader: true,
+  optionRenderer: opt => opt.label,
+  valueRenderer: opt => opt.label,
```
Comment (ascott) [inline superset/assets/javascripts/explore/components/controls/SelectControl.jsx]: should this be `valueRenderer: opt => opt.value`?

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 3]
```diff
@@ -18,6 +19,48 @@ const ROW_LIMIT_OPTIONS = [10, 50, 100, 250, 500, 1000, 5000, 10000, 50000];
 
 const SERIES_LIMITS = [0, 5, 10, 25, 50, 100, 500];
 
+const metricRenderer = m => (
```
Comment (ascott) [inline superset/assets/javascripts/explore/stores/controls.jsx]: ideally components shouldn't live in a stores file. these should probably be their own component files so they can be used in other places as needed.

File: superset/assets/javascripts/explore/components/ChartContainer.jsx (hunk 1 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explore/components/ChartContainer.jsx]: good catch, should be consistent and `props.` all around

File: superset/assets/javascripts/explore/components/controls/SelectControl.jsx (hunk 2 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explore/components/controls/SelectControl.jsx]: This is a bit confusing but essentially the `optionRenderer` applies to what's in the dropdown and the `valueRenderer` applies to the selected item(s). It's nice to have this level of control, for instance we could show the `expression` behind the metric only on the selected metric, not in the list if we wanted to.

So they both should show the label.

File: superset/assets/javascripts/explore/stores/controls.jsx (hunk 3 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: Agreed, I thought about it when starting to write it, especially makes sense as it is fairly large. I'll address that.

**Comment by ascott** [comment]
LGTM!

**Pull request merged**